central/views/platformcve/view_impl.go

```diff
@@ -0,0 +1,175 @@
+}
+
+func (v *platformCVECoreViewImpl) GetClusterIDs(ctx context.Context, q *v1.Query) ([]string, error) {
+	var err error
+	q, err = common.WithSACFilter(ctx, resources.Cluster, q)
+	if err != nil {
+		return nil, err
+	}
+
+	q.Selects = []*v1.QuerySelect{
+		search.NewQuerySelect(search.ClusterID).Distinct().Proto(),
+	}
+
+	var results []*clusterResponse
+	results, err = pgSearch.RunSelectRequestForSchema[clusterResponse](ctx, v.db, v.schema, q)
+	if err != nil || len(results) == 0 {
+		return nil, err
+	}
+
+	ret := make([]string, 0, len(results))
+	for _, r := range results {
+		ret = append(ret, r.ClusterID)
+	}
+	return ret, nil
+}
+
+func withSelectQuery(q *v1.Query) *v1.Query {
+	cloned := q.Clone()
+	cloned.Selects = []*v1.QuerySelect{
+		search.NewQuerySelect(search.CVE).Proto(),
+		search.NewQuerySelect(search.CVEID).Proto(),
+		search.NewQuerySelect(search.CVEType).Proto(),
+		search.NewQuerySelect(search.CVSS).Proto(),
+		search.NewQuerySelect(search.CVECreatedTime).Proto(),
+		search.NewQuerySelect(search.ClusterID).AggrFunc(aggregatefunc.Count).Distinct().Proto(),
+		search.NewQuerySelect(search.ClusterID).
+			Distinct().
+			AggrFunc(aggregatefunc.Count).
+			Filter("fixable_cluster_count",
+				search.NewQueryBuilder().AddBools(search.ClusterCVEFixable, true).ProtoQuery(),
+			).Proto(),
+	}
+	cloned.Selects = append(cloned.Selects, withCountByPlatformTypeSelectQuery(q).Selects...)
+	cloned.GroupBy = &v1.QueryGroupBy{
+		Fields: []string{search.CVEID.String()},
+	}
+	return cloned
+}
+
+func withCountByPlatformTypeSelectQuery(q *v1.Query) *v1.Query {
+	cloned := q.Clone()
+	cloned.Selects = append(cloned.Selects,
+		search.NewQuerySelect(search.ClusterID).
+			Distinct().
+			AggrFunc(aggregatefunc.Count).
+			Filter("generic_cluster_count",
+				search.NewQueryBuilder().
+					AddExactMatches(
+						search.ClusterPlatformType,
+						storage.ClusterType_GENERIC_CLUSTER.String(),
+					).ProtoQuery(),
+			).Proto(),
+		search.NewQuerySelect(search.ClusterID).
+			Distinct().
+			AggrFunc(aggregatefunc.Count).
+			Filter("kubernetes_cluster_count",
+				search.NewQueryBuilder().
+					AddExactMatches(
+						search.ClusterPlatformType,
+						storage.ClusterType_KUBERNETES_CLUSTER.String(),
+					).ProtoQuery(),
+			).Proto(),
+		search.NewQuerySelect(search.ClusterID).
+			Distinct().
+			AggrFunc(aggregatefunc.Count).
+			Filter("openshift_cluster_count",
+				search.NewQueryBuilder().
+					AddExactMatches(
+						search.ClusterPlatformType,
+						storage.ClusterType_OPENSHIFT_CLUSTER.String(),
+					).ProtoQuery(),
+			).Proto(),
+		search.NewQuerySelect(search.ClusterID).
+			Distinct().
+			AggrFunc(aggregatefunc.Count).
+			Filter("openshift4_cluster_count",
+				search.NewQueryBuilder().
+					AddExactMatches(
+						search.ClusterPlatformType,
+						storage.ClusterType_OPENSHIFT4_CLUSTER.String(),
+					).ProtoQuery(),
+			).Proto(),
+	)
+	return cloned
+}
+
+func withCountQuery(q *v1.Query) *v1.Query {
```

Comment:
https://github.com/stackrox/stackrox/blob/b75e0acab8600b0b76247744c2d27acd126b512a/central/views/imagecve/view_impl.go#L330
Would you like to dedupe and move it to common?

Reply:
Good idea. Moved it to common.